Will `default async fn` be supported?

Hi, me again. Now the trait with default async functions will not be support, say:

``` rust
use async_trait::async_trait;
use futures::executor::block_on;
pub struct Something<'a> {
    name: &'a str,
}

#[async_trait]
pub trait Abstract<'a> {
    async fn hello(thing: Something<'a>) -> String;
    async fn hello_twice(thing1: Something<'a>, thing2: Something<'a>) -> String {
        let str1 = Self::hello(thing1).await;
        let str2 = Self::hello(thing2).await;
        str1 + str2.as_ref()
    }
}

pub struct Real();
#[async_trait]
impl <'a> Abstract<'a> for Real {
    async fn hello(thing: Something<'a>) -> String {
        String::from(thing.name)
    }
}

fn main() {
    let thing1 = Something{ name: "hello" };
    let thing2 = Something {name: "world"};
    let name = block_on(Real::hello_twice(thing1, thing2));
    println!("name: {}", name);
}
```

Will get error:
```
error[E0412]: cannot find type `AsyncTrait` in this scope
 --> src\main.rs:7:1
  |
7 | #[async_trait]
  | ^^^^^^^^^^^^^^ not found in this scope
```

--- a/src/receiver.rs
+++ b/src/receiver.rs
@@ -19,6 +19,11 @@ pub fn has_self_in_block(block: &mut Block) -> bool {
 struct HasSelf(bool);
 
 impl VisitMut for HasSelf {
+    fn visit_expr_path_mut(&mut self, expr: &mut ExprPath) {
+        self.0 |= expr.path.segments[0].ident == "Self";
+        visit_mut::visit_expr_path_mut(self, expr);
+    }
+
     fn visit_type_path_mut(&mut self, ty: &mut TypePath) {
         self.0 |= ty.path.segments[0].ident == "Self";
         visit_mut::visit_type_path_mut(self, ty);

--- a/tests/test.rs
+++ b/tests/test.rs
@@ -341,3 +341,22 @@ mod issue28 {
         async fn h(); // do not chain
     }
 }
+
+// https://github.com/dtolnay/async-trait/issues/31
+pub mod issue31 {
+    use async_trait::async_trait;
+
+    pub struct Struct<'a> {
+        pub name: &'a str,
+    }
+
+    #[async_trait]
+    pub trait Trait<'a> {
+        async fn hello(thing: Struct<'a>) -> String;
+        async fn hello_twice(one: Struct<'a>, two: Struct<'a>) -> String {
+            let str1 = Self::hello(one).await;
+            let str2 = Self::hello(two).await;
+            str1 + &str2
+        }
+    }
+}

